assets/js/blocks/product-categories/block.js

```diff
@@ -36,6 +22,10 @@ class ProductCategoriesBlock extends Component {
 		this.renderOptions = this.renderOptions.bind( this );
 	}
 
+	componentWillMount() {
+		this.id = uniqueID( '', 'wc-block-product-categories' );
```

Comment:
Can we move this to the constructor? Using `componentWillMount` is discouraged:
https://reactjs.org/docs/react-component.html#unsafe_componentwillmount